### ISSUE TITLE
Files not updating inside container after updating from 4.29.0 to 4.30.0

### ISSUE THREAD

**Issue by 8ctopus** [docker/for-win#14060]
### Description

I have a Windows setup for development using the Hyper-V engine instead of WSL2 for inotify to work.

When a file is updated in the Windows file system, it's not automatically updated inside the container since 4.30.0. (The file is in the same state as when the container was started.)

Reverting to 4.29.0 fixed the problem.

### Reproduce

- start a container with shared folder
- edit in windows a file in the shared folder
- the file is not changed in the container

### Expected behavior

When a file is changed in the windows filesystem, it should get updated inside the container.

### docker version

```bash
Client:
 Cloud integration: v1.0.35+desktop.13
 Version:           26.1.1
 API version:       1.45
 Go version:        go1.21.9
 Git commit:        4cf5afa
 Built:             Tue Apr 30 11:48:43 2024
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.30.0 (149282)
 Engine:
  Version:          26.1.1
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.9
  Git commit:       ac2de55
  Built:            Tue Apr 30 11:48:28 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.31
  GitCommit:        e377cd56a71523140ca6ae87e30244719194a521
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    26.1.1
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.14.0-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.27.0-desktop.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.29
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.23
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.8.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 2
  Running: 2
  Paused: 0
  Stopped: 0
 Images: 2
 Server Version: 26.1.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: e377cd56a71523140ca6ae87e30244719194a521
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
  cgroupns
 Kernel Version: 6.6.26-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 1.92GiB
 Name: docker-desktop
 ID: 561f1203-4318-48df-ad63-6796ec8c0eca
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=npipe://\\.\pipe\docker_cli
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

5FC51C8F-9C92-4D9D-910E-CC14DB38403E/20240509120112

### Additional Info

Very similar to this older issue: https://github.com/docker/for-win/issues/5530

**Comment by fikusz01** [comment]
Same problem here. Just upgraded to v4.30.0

**Comment by AngelCifuentes** [comment]
Same issue here in multiple configs (Windows 10/11, Mac docker client).
Folder appear Mounted and with privileges but, when make changes in host, aply it in swap file tmp, not real file, not sync.

I'll rollback to 4.29.0 and all ok. Try with more configs, caché flush, other volumes, images, etc and add ":consistent" tag to volume, nothing work, only rollback to 4.29.0.

To rollback:
Download 4.29 for your OS from link
https://docs.docker.com/desktop/release-notes/#4290

Open terminal:
.\ Docker_installer_4_29_0_path.exe install --disable-version-check

**Comment by pneme** [comment]
Same problem.

**Comment by berkanumutlu** [comment]
I have same problem. I rebuild containers, start older containers(already builded and using containers before Docker v4.30.0) and later tried that comment https://github.com/docker/for-win/issues/5530#issuecomment-577634366 in this older issue: https://github.com/docker/for-win/issues/5530 but not worked. So I had to downgrade Docker version to v4.29.0 for temporary solution. https://github.com/docker/for-win/issues/13207#issuecomment-1411903285

**Comment by nirvana72** [comment]
same problem, and how to roll back to 429 safely? uninstall 430 will lost my container

**Comment by berkanumutlu** [comment]
> same problem, and how to roll back to 429 safely? uninstall 430 will lost my container

I didn't uninstall Docker, just downloaded v4.29.0 from this [link](https://docs.docker.com/desktop/release-notes/#4290) and tried running the installer with from the CLI `<installer_file_path>/DockerInstaller.exe install --disable-version-check` and I didn't lose the containers.

**Comment by caelus-dev** [comment]
Same problem here.

> > same problem, and how to roll back to 429 safely? uninstall 430 will lost my container
> 
> I didn't uninstall Docker, just downloaded v4.29.0 from this [link](https://docs.docker.com/desktop/release-notes/#4290) and tried running the installer with from the CLI `<installer_file_path>/DockerInstaller.exe install --disable-version-check` and I didn't lose the containers.

Are there any way to run a similar command on Windows? `--disable-version-check` argument doesn't work for me on Win.

**Comment by 8ctopus** [comment]
> > same problem, and how to roll back to 429 safely? uninstall 430 will lost my container
> 
> I didn't uninstall Docker, just downloaded v4.29.0 from this [link](https://docs.docker.com/desktop/release-notes/#4290) and tried running the installer with from the CLI `<installer_file_path>/DockerInstaller.exe install --disable-version-check` and I didn't lose the containers.

correct, you don't loose containers if you install over the previous version

**Comment by 8ctopus** [comment]
> Same problem here.
> 
> > > same problem, and how to roll back to 429 safely? uninstall 430 will lost my container
> > 
> > 
> > I didn't uninstall Docker, just downloaded v4.29.0 from this [link](https://docs.docker.com/desktop/release-notes/#4290) and tried running the installer with from the CLI `<installer_file_path>/DockerInstaller.exe install --disable-version-check` and I didn't lose the containers.
> 
> Are there any way to run a similar command on Windows? `--disable-version-check` argument doesn't work for me on Win.

It worked for me, what error did you get?

**Comment by caelus-dev** [comment]
> > Same problem here.
> > > > same problem, and how to roll back to 429 safely? uninstall 430 will lost my container
> > > 
> > > 
> > > I didn't uninstall Docker, just downloaded v4.29.0 from this [link](https://docs.docker.com/desktop/release-notes/#4290) and tried running the installer with from the CLI `<installer_file_path>/DockerInstaller.exe install --disable-version-check` and I didn't lose the containers.
> > 
> > 
> > Are there any way to run a similar command on Windows? `--disable-version-check` argument doesn't work for me on Win.
> 
> It worked for me, what error did you get?

Okay, It works, I just missed an apostrophe in the file name.

**Comment by BoGnY** [comment]
same problem, only downgrade solved it

**Comment by berkut1** [comment]
the same problem, you can't update symfony cache. You need always restart the whole docker after updating cache.
P.S Docker with Hyper-V
P.S2 Downgrade to 4.29 solved it too

**Comment by dt-ebermudez** [comment]
Same problem here. Just upgraded to v4.30.0

only downgrade to v4.29.0 solved it

**Comment by 8ctopus** [comment]
Looks like the Docker team doesn't care about Windows users...

**Comment by K2ouMais** [comment]
This is so annoying... It makes impossible to develop right now, when everything is in Docker...

I am just developing my app and all the changes dont show up, unless I restart the containers.

Will also downgrade ;(

**Comment by AngelCifuentes** [comment]
> This is so annoying... It makes impossible to develop right now, when everything is in Docker...
> 
> I am just developing my app and all the changes dont show up, unless I restart the containers.
> 
> Will also downgrade ;(

I'll try a lot of solutions and only downgrade to 4.29 its ok for my develops.
With WSL2 in Windows containers and tag :consistent not reproduce this issue but in other scenarios (with HVM), caché persist...

I'll try move Win to Linux containers (with move all data...) and try if this issue its solved in this scenario

**Comment by nilladin** [comment]
Same prob. here on W10 with Hyper-V... WSL-2 works good but is too slow.

I'll try to downgrade, hope it will help.

**Comment by dynamicnet** [comment]
Same issue, downgrade get the job done.
Downgrade must be launched using command line, your containers should stay in place.

`DockerDesktopInstaller.exe install --disable-version-check`

**Comment by Sammaye** [comment]
Really wish downgrade was in installer, I only learned of this after losing all my work.... and of course same problem as all others, 4.29 works fine

Also installing 4.29 does not require my virus scanner to be closed, while installing 4.30 does... something not right about that

**Comment by fikusz01** [comment]
Btw 4.29 is affected with another annoying bug: it randomly but surely hangs up containers, only docker desktop restart helps. Yesterday found that open ticket also, and still no solution.

**Comment by BoGnY** [comment]
> Btw 4.29 is affected with another annoying bug: it randomly but surely hangs up containers, only docker desktop restart helps. Yesterday found that open ticket also, and still no solution.

this bug started more or less from 4.24 😢

**Comment by fikusz01** [comment]
> this bug started more or less from 4.24 😢

Today I installed 4.24.0, just working fine. Gonna stick to that forever :) Thanks!

**Comment by Fosol** [comment]
4.29.0 worked for me on Windows, but it required a few attempts.  I had to reconfigure file shares and disable updates.

**Comment by thienanblog** [comment]
> Looks like the Docker team doesn't care about Windows users...

I have the same problem on Mac with Laradock!

**Comment by OldSneerJaw** [comment]
I also ran into this problem using a named volume in Docker Compose on MacOS 13 (Ventura) with an Intel CPU. As others mentioned only when I fully restarted Docker Desktop would file changes be reflected in a running container. After downgrading Docker Desktop from 4.30.0 to 4.29.0 it all worked again without requiring restarts of Docker Desktop.

**Comment by LeoNguyenHQ** [comment]
Same issue here. Downgrading to 4.29 worked.

`Docker Desktop Installer.exe install --disable-version-check`

**Comment by BoGnY** [comment]
> > this bug started more or less from 4.24 😢
> 
> Today I installed 4.24.0, just working fine. Gonna stick to that forever :) Thanks!

pay attention.. in the related issue I found this problem was reported starting from 4.19 😢 

EDIT: many reported that the last working version is 4.25.2 👍🏻

**Comment by 8ctopus** [comment]
> Btw 4.29 is affected with another annoying bug: it randomly but surely hangs up containers, only docker desktop restart helps. Yesterday found that open ticket also, and still no solution.

Thank you for sharing, I also experience this bug and I was wondering which version it came from.

**Comment by ezekel** [comment]
same problem I just rollback to 4.29

**Comment by Myke79** [comment]
exactly the same here, wasted hours to find and solve the problem, only downgrade to 4.29 helped.

**Comment by 8ctopus** [comment]
2 weeks since the bug was reported and it doesn't even have someone from the docker team assigned...

**Comment by MihaelaStoica** [comment]
Thanks for reporting the issue and apologies for the inconvenience it's caused and for the belated response.
We are working on a fix and will update here as soon as we have more news.

**Comment by brandonlenz** [comment]
Same problem on mac os sonoma.

docker compose services with mounted volumes defined do not recognize file system changes on the host unless docker desktop is restarted. (stopping, rebuilding, and starting containers is not enough)

**Comment by walter-developer** [comment]
I have the same problem synchronizing files from Windows to the Linux container on my project volumes.

**Comment by sebsobseb** [comment]
Same problem on mac os Monterey Apple M1 Max
downgraded to 4.29

**Comment by chludwig** [comment]
Here the same problem. Files don't get synced until Docker Desktop is restarted (Microsoft Windows 11 Pro, Version 10.0.22631 Build 22631).

**Comment by batcats** [comment]
With the removal/replacement of the working mutagen plugin for Windows + Docker Desktop + WSL2 it has started. Integrating the feature into Windows + Docker Desktop has removed WSL2 and forced a switch to the unstable and slower Hyper-V if you rely on file sharing. With each new version, file sharing has gotten worse. Until this version. Too bad.

**Comment by ravi-palaniappan-navico** [comment]
+1 Same problem and quite annoying for couple of hours to find out... downgraded to 4.29.0..

**Comment by nicholascarvalho** [comment]
I've tried using` :consistent` on the volumes and a few other suggestions but the only thing that fixed this for me was also a downgrade to 4.29. I'm on windows 11 with linux volumes and images.

**Comment by DoctorMcKay** [comment]
It seems crazy to me that 4.30.0 for Windows hasn't been pulled at the very least.

**Comment by BigOHenry** [comment]
Same here, rollback to 4.29.0 fixed the issue.

**Comment by isntw** [comment]
Same problem here, it's so frustrating

**Comment by ctberdon** [comment]
any updates if this will be fixed?

**Comment by pdeberranger** [comment]
Same for me, downgrade is the only issue!

**Comment by w-A-L-L-e** [comment]
Same problem on macos sonoma with 4.2.30. Downgrade to 4.2.29 fixes that issue but has different issue that containers go to sleep and don't wake up if inactive for a while. 
When this happens you can't stop/start the containers either you also have to restart entire docker app:
```
(HTTP code 500) server error - cannot stop container: ebac4d9da09c28734cb46d21e30cd301ed5d04dff5410314d9ff36922475ceac: tried to kill container, but did not receive an exit event
```

**Comment by 8ctopus** [comment]
@w-A-L-L-e Same issue on Windows. If I go for lunch and put the computer to sleep, when I turn it back on, the docker containers are no longer working. Have to restart docker desktop and the containers.

**Comment by Myke79** [comment]
That's another issue described here: 
https://github.com/docker/desktop-feedback/issues/113

And that one is still unsolved since 4.26.0.
The only solution to downgrade to 4.25.2 and not upgrading, until the linked issue and this one is solved.

**Comment by josepdecid** [comment]
I faced the same problem when updating to 4.30.0. Rolling back to 4.29.0 fixes the problem.
On Mac, just download the [4.29.0 release](https://docs.docker.com/desktop/release-notes/#4290) and replace the current installation.

**Comment by Webinx** [comment]
Any ETA on fixing the issue? Prior versions are also affected by very annoying issues that were fixed in version 4.30. This should have the highest priority.

**Comment by RichardForshaw** [comment]
Upgraded recently and also finding the same issue. Surprised it is a month old. Watching for fix.

**Comment by MihaelaStoica** [comment]
Thank you all for your patience on this and apologies again for the issue. We will include a fix in Docker Desktop 4.31 which is due to be released this week.

**Comment by hahatulk** [comment]
Same problem here

**Comment by x0rsw1tch** [comment]
🤞 Hope that this fixes [#13894](https://github.com/docker/desktop-feedback/issues/113), been stuck on 4.25.2 for a while now

**Comment by AngelCifuentes** [comment]
With this patch (4.31.0) solved the file updating problem docker/for-win#14060 but not docker/desktop-feedback#113 , needed to close and open docker cli again to work

Tested in:
- Windows 10 / 11 laptops (Dell, Hp, etc)
- Hyper V

Last versión that this work ok is 4.25.2

**Comment by maxcleme** [comment]
Closing this issue because a fix has been released in Docker Desktop `4.31.0`. See the [release notes](https://docs.docker.com/desktop/release-notes/#4310) for more details.
